### PR TITLE
Allow @ and : in smb password

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

**Comment by marsaud** [comment]
This is not a very great fix, but we may soon rework the whole stuff for #1 , with other choices, regex perhaps

File: src/index.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ export const parse = (remote) => {
     remote.share = share
   } else if (type === 'smb') {
     remote.type = 'smb'
-    const [auth, smb] = rest.split('@')
+    const parts = rest.split('@')
+    const smb = parts.pop()
+    const auth = parts.join('@')
```
Comment (julien-f) [inline src/index.js]: Using `rest.slice(0, rest.lastIndexOf('@'))` would be a bit better.

**Comment by julien-f** [comment]
Also, needs rebase.

File: src/index.js [hunk 2]
```diff
@@ -18,7 +18,8 @@ export const parse = (remote) => {
     remote.share = share
   } else if (type === 'smb') {
     remote.type = 'smb'
-    const [auth, smb] = rest.split('@')
+    const smb = rest.slice(rest.lastIndexOf('@'))
+    const auth = rest.slice(0, rest.lastIndexOf('@'))
```
Comment (julien-f) [inline src/index.js]: It's ok for this time, but in the future, use a variable to store the index and avoid unnecessary calls.

**Comment by julien-f** [comment]
Does this PR also fixes the use of `:`?

**Comment by marsaud** [comment]
No, if this is not enough for now, I rewrite the whole stuff with a regex ... or maybe not.

**Comment by julien-f** [comment]
Thanks :)

**Pull request merged**